src/scripts/main.js

```diff
@@ -1,3 +1,273 @@
 'use strict';
 
-// write your code here
+document.addEventListener('DOMContentLoaded', () => {
+  const gridDisplay = document.querySelector('.grid');
+  const scoreDisplay = document.getElementById('score');
+  const resultDisplay = document.getElementById('result');
+  const newGameButton = document.getElementById('new-game');
+  const squares = [];
+  const width = 4;
+  let score = 0;
+
+  function resetGame() {
+    for (let i = 0; i < squares.length; i++) {
+      squares[i].innerHTML = 0;
+    }
+    score = 0;
+    scoreDisplay.innerHTML = score;
+    resultDisplay.innerHTML = '';
+    generate();
+    generate();
+  }
+
+  function clearStyles() {
+    for (let i = 0; i < squares.length; i++) {
+      squares[i].style.backgroundColor = '';
+    }
+  }
+
+  function handleNewGameClick() {
+    location.reload();
+    resetGame();
+    clearStyles();
+    addColours();
+    document.addEventListener('keyup', control);
+  }
+
+  newGameButton.addEventListener('click', handleNewGameClick);
+
+  function createBoard() {
+    for (let i = 0; i < width * width; i++) {
+      const square = document.createElement('div');
+
+      square.innerHTML = 0;
+      gridDisplay.appendChild(square);
+      squares.push(square);
+    }
+    generate();
+    generate();
+  }
+  createBoard();
+
+  function generate() {
+    let randomNumber = Math.floor(Math.random() * squares.length);
+
+    while (squares[randomNumber].innerHTML !== '0') {
+      randomNumber = Math.floor(Math.random() * squares.length);
+    }
+
+    squares[randomNumber].innerHTML = '2';
+    checkForGameOver();
+  }
+
+  function moveRight() {
+    for (let i = 0; i < 16; i++) {
```

Comment:
16 should be saved in some variable,  so we can avoid using [magic numbers](https://www.pluralsight.com/tech-blog/avoiding-magic-numbers/) look through your code and fix other cases